tenant_security_request.go

```diff
@@ -145,23 +148,24 @@ func (r *tenantSecurityRequest) parseAndDoRequest(endpoint *tspEndpoint, request
 // doRequest sends a JSON request body to a TSP endpoint and returns the response body bytes.
 // If the request can't be sent, or if the server response code indicates an error, this function
 // returns an error instead.
-func (r *tenantSecurityRequest) doRequest(endpoint *tspEndpoint, reqBody io.ReadCloser) ([]byte, error) {
+func (r *tenantSecurityRequest) doRequest(ctx context.Context,
+	endpoint *tspEndpoint,
+	reqBody io.Reader) ([]byte, error) {
 	// Build the request.
 	url := r.tspAddress.ResolveReference((*url.URL)(endpoint))
-	req := http.Request{
-		URL:    url,
-		Method: http.MethodPost,
-		Body:   reqBody,
-		Header: map[string][]string{
-			"User-Agent":    {fmt.Sprintf("Tenant Security Client Go v%s", Version)},
-			"Content-Type":  {"application/json"},
-			"Accept":        {"application/json"},
-			"Authorization": {fmt.Sprintf("cmk %s", r.apiKey)},
-		},
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url.String(), reqBody)
```

Comment:
If this is just going to take the URL's string (and I assume do validation as part of building the request), should we just have `const`s with the url strings at the top of the file? Instead of our current `var`/`init` method

Reply:
Yeah, the http API's use of strongly typed URL vs. weakly typed string seems schizophrenic. I like stronger types, but it seems like unnecessary work in this context. Do you think the extra type safety is worth the extra work?

Reply:
I went down this path and didn't like how it turned out, so I think I'm happy with the safety we currently have